index.mjs

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+// Bootstraps yargs in ESM mode:
+import escalade from 'escalade/sync'
+import { fileURLToPath } from 'url';
+import Parser from 'yargs-parser'
+import { dirname, resolve } from 'path'
+import { YargsFactory } from './build/lib/yargs-factory.js'
+
+// TODO: stop using createRequire as soon as we've ported: cliui, y18n, etc.
+import { createRequire } from 'module';
```

Comment:
@QmarkC if once we port `y18n` and `cliui` we'll be getting darn close to an ESM version of yargs.

Note that `createRequire` is Node.js specific, so won't port well to other runtimes.